core/chains/evm/gas/fee_history_estimator.go

```diff
@@ -0,0 +1,414 @@
+	nextBlock := big.NewInt(0).Add(feeHistory.OldestBlock, big.NewInt(int64(u.config.BlockHistorySize)))
+
+	priorityFee := big.NewInt(0)
+	priorityFeeThreshold := big.NewInt(0)
+	for _, fee := range feeHistory.Reward {
+		priorityFee = priorityFee.Add(priorityFee, fee[0])
+		// We don't need an average, we need the max value
+		priorityFeeThreshold = bigmath.Max(priorityFeeThreshold, fee[1])
+	}
+	priorityFeeThresholdWei := assets.NewWei(priorityFeeThreshold)
+
+	u.priorityFeeThresholdMu.Lock()
+	u.priorityFeeThreshold = priorityFeeThresholdWei
+	u.priorityFeeThresholdMu.Unlock()
+
+	// eth_feeHistory may return less results than BlockHistorySize so we need to divide by the length of the result
+	maxPriorityFeePerGas := assets.NewWei(priorityFee.Div(priorityFee, big.NewInt(int64(len(feeHistory.Reward)))))
+	// baseFeeBufferPercentage is used as a safety to catch fluctuations in the next block.
+	maxFeePerGas := baseFee.AddPercentage(BaseFeeBufferPercentage).Add(maxPriorityFeePerGas)
+
+	promFeeHistoryEstimatorBaseFee.WithLabelValues(u.chainID.String()).Set(float64(baseFee.Int64()))
+	promFeeHistoryEstimatorMaxPriorityFeePerGas.WithLabelValues(u.chainID.String()).Set(float64(maxPriorityFeePerGas.Int64()))
+	promFeeHistoryEstimatorMaxFeePerGas.WithLabelValues(u.chainID.String()).Set(float64(maxFeePerGas.Int64()))
+
+	u.logger.Debugf("Fetched new dynamic prices, nextBlock#: %v - oldestBlock#: %v - maxFeePerGas: %v - maxPriorityFeePerGas: %v - maxPriorityFeeThreshold: %v",
+		nextBlock, feeHistory.OldestBlock, maxFeePerGas, maxPriorityFeePerGas, priorityFeeThresholdWei)
+
+	u.dynamicPriceMu.Lock()
+	defer u.dynamicPriceMu.Unlock()
+	u.dynamicPrice.FeeCap = maxFeePerGas
+	u.dynamicPrice.TipCap = maxPriorityFeePerGas
+	return u.dynamicPrice, nil
+}
+
+func (u *FeeHistoryEstimator) getDynamicPrice() (fee DynamicFee, err error) {
+	u.dynamicPriceMu.RLock()
+	defer u.dynamicPriceMu.RUnlock()
+	if u.dynamicPrice.FeeCap == nil || u.dynamicPrice.TipCap == nil {
+		return fee, fmt.Errorf("dynamic price not set")
+	}
+	return u.dynamicPrice, nil
+}
+
+// BumpLegacyGas provides a bumped gas price value by bumping the previous one by BumpPercent.
+// If the original value is higher than the max price it returns an error as there is no room for bumping.
+// It aggregates the market, bumped, and max gas price to provide a correct value.
+func (u *FeeHistoryEstimator) BumpLegacyGas(ctx context.Context, originalGasPrice *assets.Wei, gasLimit uint64, maxPrice *assets.Wei, _ []EvmPriorAttempt) (*assets.Wei, uint64, error) {
+	// Sanitize original fee input
+	if originalGasPrice == nil || originalGasPrice.Cmp(maxPrice) >= 0 {
+		return nil, 0, fmt.Errorf("%w: error while retrieving original gas price: originalGasPrice: %s. Maximum price configured: %s",
+			commonfee.ErrBump, originalGasPrice, maxPrice)
+	}
+
+	currentGasPrice, err := u.getGasPrice()
+	if err != nil {
+		return nil, 0, err
+	}
+
+	bumpedGasPrice := originalGasPrice.AddPercentage(u.config.BumpPercent)
```

Comment:
Shouldn't we also consider `BumpMin` here? Or in `LimitBumpedFee`?

Reply:
This is on purpose. `BumpMin` is hard to define, which can cause problems to the user and, as of now, just gets copy pasted. We need to deprecate configs that don't provide much value in order to reduce the chances of misconfiguration. Having a `BumpPercent` of more than 10%, which is mandatory, almost always covers the `BumpMin` case, and in the few edge cases this can happen we throw an error instead, i.e. prices lower than 10 wei.

Reply:
I guess for the initial PR that's fine. Just to cover our bases when we start using this estimator, we should double check with Ship if that config was really copy pasted or if there's some weird chain behavior for any of the ones that have it set currently.